### PR TITLE
Logging rewrite 2

### PR DESCRIPTION
**Pull request by FredKSchott**
**Note:** Waiting to enter `4.0` beta period before merging this PR. If it gets out of date with `main`, it can be rebased (by @FredKSchott if the merge conflicts are non-trivial).

## Changes

A re-attempt at https://github.com/withastro/astro/pull/8833, scoped down.  
See this comment for context: https://github.com/withastro/astro/pull/8833#issuecomment-1811977053

- A complete, holistic audit of all messages logged by the Astro CLI
- Clarity: rewrote some info/warn/error messages to be more clear and useful
- Consistency: reduced differences in logged icons, color, formatting, phrasing. For example, the color cyan is now (almost) always used for links. 
- Focus: removed some noisy startup logs that weren't useful
- Focus: removed distracting debug `[label]` information from most logs
- ~~Better Errors:~~ These changes introduced in https://github.com/withastro/astro/pull/8833 have been reverted, and are no longer included in this PR.
- Removed the unnecessary `(x2)`, `(x3)`, etc. effect in the dev server
- Added consistent request logging (including render time)

## Videos

https://github.com/withastro/astro/assets/622227/0f8c8034-f1e5-4696-bc95-e8235c0ffe78

https://github.com/withastro/astro/assets/622227/4a5b16b5-05d4-4ec8-8d25-c094e9391bd8

_Note: All error-related changes have been removed from this PR and are out-of-date in the videos above._

### Example: `(xN)` Formatting Weirdness

Rewriting the CLI output would cause weird formatting issues sometimes. In this example, resizing the CLI window did it.
Screen Shot 2023-10-13 at 2 13 24 PM

#### Before

Screen Shot 2023-10-13 at 2 13 52 PM

#### After

Screen Shot 2023-10-13 at 2 14 52 PM

### Build Output

Mainly stylistic, colors changed intentionally. 

#### Before
Screen Shot 2023-10-13 at 2 27 41 PM

#### After
Screen Shot 2023-10-13 at 2 18 16 PM

## Testing

- Tests updated (caught a small test error in the process)

## Docs

- N/A (to confirm with docs)

### CODE REVIEW

File: packages/astro/src/core/build/index.ts [hunk 1]
```diff
@@ -157,10 +158,11 @@ class AstroBuilder {
...
 
-		this.logger.info('build', `output target: ${colors.green(this.settings.config.output)}`);
-		if (this.settings.adapter) {
-			this.logger.info('build', `deploy adapter: ${colors.green(this.settings.adapter.name)}`);
-		}
+		this.logger.info('build', `output: ${blue('"' + this.settings.config.output + '"')}`);
+		this.logger.info('build', `outDir: ${blue(fileURLToPath(this.settings.config.outDir))}`);
+		// if (this.settings.adapter) {
+		this.logger.info('build', `adapter: ${blue('@vercel/serverless')}`);
+		// }
```
Comment (Princesseuh) [inline packages/astro/src/core/build/index.ts:165]: Oh no, Fred is forcing SSR on all of us

File: packages/astro/src/core/create-vite.ts [hunk 2]
```diff
@@ -131,7 +142,6 @@ export async function createVite(
 			astroScriptsPageSSRPlugin({ settings }),
 			astroHeadPlugin(),
 			astroScannerPlugin({ settings, logger }),
-			astroInjectEnvTsPlugin({ settings, logger, fs }),
```
Comment (Princesseuh) [inline packages/astro/src/core/create-vite.ts:134]: Not intended, I assume

File: packages/astro/src/core/build/index.ts (hunk 1 above)
Comment (FredKSchott) [reply packages/astro/src/core/build/index.ts:165]: Ha! Good catch, will revert

File: packages/astro/src/core/create-vite.ts (hunk 2 above)
Comment (FredKSchott) [reply packages/astro/src/core/create-vite.ts:134]: Hmm, took a deeper look to refresh my memory and no longer see any code running this besides `sync`. Reverting this!

**Review by matthewp — APPROVED** [review]

File: packages/astro/src/core/build/index.ts [hunk 3]
```diff
@@ -157,9 +158,10 @@ class AstroBuilder {
 		await runHookBuildStart({ config: this.settings.config, logging: this.logger });
 		this.validateConfig();
 
-		this.logger.info('build', `output target: ${colors.green(this.settings.config.output)}`);
+		this.logger.info('build', `output: ${blue('"' + this.settings.config.output + '"')}`);
+		this.logger.info('build', `outDir: ${blue(fileURLToPath(this.settings.config.outDir))}`);
 		if (this.settings.adapter) {
-			this.logger.info('build', `deploy adapter: ${colors.green(this.settings.adapter.name)}`);
+		  this.logger.info('build', `adapter: ${blue('@vercel/serverless')}`);
```
Comment (Princesseuh) [inline packages/astro/src/core/build/index.ts:164]: Presumably this is supposed to be the adapter the user is using.

Comment (FredKSchott) [reply packages/astro/src/core/build/index.ts:164]: that's what I get for trying to fix this quickly from the web UI on my phone... thanks for the thorough review on this @Princesseuh

**Review by bluwy — APPROVED** [review]
Like the new logs! I've create a new [`next` branch](https://github.com/withastro/astro/tree/next) that you can merge this into now

File: packages/astro/src/core/build/index.ts [hunk 4]
```diff
@@ -157,9 +158,10 @@ class AstroBuilder {
 		await runHookBuildStart({ config: this.settings.config, logging: this.logger });
 		this.validateConfig();
 
-		this.logger.info('build', `output target: ${colors.green(this.settings.config.output)}`);
+		this.logger.info('build', `output: ${blue('"' + this.settings.config.output + '"')}`);
+		this.logger.info('build', `directory: ${blue(fileURLToPath(this.settings.config.outDir))}`);
 		if (this.settings.adapter) {
-			this.logger.info('build', `deploy adapter: ${colors.green(this.settings.adapter.name)}`);
+		  this.logger.info('build', `adapter: ${colors.green(this.settings.adapter.name)}`);
```
Comment (Princesseuh) [inline packages/astro/src/core/build/index.ts:164]:
```suggestion
		  this.logger.info('build', `adapter: ${green(this.settings.adapter.name)}`);
```
I think?

**Comment by FredKSchott** [comment]
tests passing! merging into `next`

**Pull request merged**